### PR TITLE
Final API

### PR DESCRIPTION
**Pull request by cibo94**
Please merge after #337

### CODE REVIEW

File: doc/api_package_builder.rst [hunk 1]
```diff
@@ -0,0 +1,6 @@
+Package builder module
+======================
+
+.. automodule:: rpg.package_builder
+    :members:
+    :undoc-members:
```
Comment (jsilhan) [inline doc/api_package_builder.rst]: we don't want to have this in API

File: doc/api_plugin_engine.rst [hunk 2]
```diff
@@ -0,0 +1,6 @@
+Plugin engine module
+====================
+
+.. automodule:: rpg.plugin_engine
+    :members:
+    :undoc-members:
```
Comment (jsilhan) [inline doc/api_plugin_engine.rst]: we don't want to have this in API

File: doc/api_project_builder.rst [hunk 3]
```diff
@@ -0,0 +1,6 @@
+Project builder module
+======================
+
+.. automodule:: rpg.project_builder
+    :members:
+    :undoc-members:
```
Comment (jsilhan) [inline doc/api_project_builder.rst]: we don't want to have this in API

File: doc/api_source_loader.rst [hunk 4]
```diff
@@ -0,0 +1,6 @@
+Source loader module
+====================
+
+.. automodule:: rpg.source_loader
+    :members:
+    :undoc-members:
```
Comment (jsilhan) [inline doc/api_source_loader.rst]: we don't want to have this in API

File: doc/index.rst [hunk 5]
```diff
@@ -18,6 +18,12 @@ Contents:
    command_ref
    writing_plugins
    api_base
+   api_command
+   api_project_builder
+   api_package_builder
+   api_source_loader
+   api_plugin_engine
+   api_plugin
```
Comment (jsilhan) [inline doc/index.rst]: `api_command` and `api_plugin` only, please

File: rpg/__init__.py [hunk 6]
```diff
@@ -20,8 +19,34 @@
...
+>>> base.target_distro = "fedora-22"
+>>> base.fetch_repos(base.target_distro, base.target_arch)
+>>> base.run_extracted_source_analysis()
+>>> base.run_patched_source_analysis()
+>>> base.build_project()
+>>> base.run_compiled_source_analysis()
+>>> base.install_project()
+>>> base.run_installed_source_analysis()
+>>> base.build_srpm()
+>>> base.build_rpm_recover(self.base.target_distro, self.base.target_arch)
```
Comment (jsilhan) [inline rpg/__init__.py]: cool

File: rpg/plugins/misc/files_to_pkgs.py [hunk 7]
```diff
@@ -5,13 +5,17 @@
 class FilesToPkgsPlugin(Plugin):
 
     TRANSLATED = {}
+    IGNORE = []
```
Comment (jsilhan) [inline rpg/plugins/misc/files_to_pkgs.py]: these should be private

File: rpg/source_loader.py [hunk 8]
```diff
@@ -8,11 +8,28 @@
...
+>>> from rpg.source_loader import SourceLoader
+>>> src_ldr = SourceLoader()
+>>> src_ldr.download_git_repo("http://www.github.com/sample_repo", "arch.tar")
+>>> src_ldr.load_sources("../arch.tar", "/tmp/rpg-51A5F8")
+    """
+
+    #: _compressions = {
+    #:   compr_extension: [compr_prog, extract_switch, output_set_switch,
+    #:    { addition_compr_ext: [compr_prog, switch] }, ...
+    #: }
```
Comment (jsilhan) [inline rpg/source_loader.py]: what is this?

File: tests/mock_build/test_build_libhif.py [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+
+
+class BuildTest(RpgTestCase):
+
+    def test_libhif(self):
+        self.base = Base()
+        self.base.sack = self.base.load_dnf_sack()
+        self.base.load_plugins()
+        self.base.load_project_from_url(
+            "https://github.com/rpm-software-management/libhif")
```
Comment (jsilhan) [inline tests/mock_build/test_build_libhif.py]: load the zip / tarball of specific version please. In a few days hawkey will be merged with libhif and autotools will be replaced with cmake

File: tests/unit/test_plugins.py [hunk 10]
```diff
@@ -165,22 +157,17 @@ def test_setuptools(self):
...
-                        "test2.2", "test3.1 >= 0.4.6", "test3.2 >= 1.7.11",
-                        "test3.3 >= 4.11.0", "test4 <= test5", "test5.1",
-                        "test5.2", "test6 > test5", "test7.1", "test7.2",
-                        "test7.3", "test8.1 = doNotKnowIfViableButItWorks",
-                        "test8.2", "test9.1", "test9.2"])
-        autotplug.patched(
+    def test_autotools_deps(self):
+        auto = AutotoolsPlugin()
+        self.spec.build_required_files = set()
+        auto.compiled(
```
Comment (jsilhan) [inline tests/unit/test_plugins.py]: great, can detect files

File: rpg/plugins/misc/files_to_pkgs.py (hunk 7 above)
Comment (cibo94) [inline rpg/plugins/misc/files_to_pkgs.py]: Yeah this is another PR but I have forgotten to reset to upstream/master when I was refactoring so this is heavelly depend on #337.

**Comment by jsilhan** [comment]
Merged, thanks

**Pull request merged**